### DISCUSSION TITLE
Add JWT authentication to a LB4 Socket App

### DISCUSSION THREAD

**Discussion by baltameme** [loopbackio/loopback-next#7179] [Q&A]
I have built a simple chat based on the example shared in the following repo:
https://github.com/raymondfeng/loopback4-example-websocket/

What i am trying to do now is closing the connection over this App so it can be allowed through specific authenticated users. I want to enquire how can i integrate the JWT authentication as part of the socket app life cycle.

**Comment by dougal83** [comment]
Hey @baltameme 

You might want to pose your question on https://stackoverflow.com/ with tags loopback and websocket.

**Comment by raymondfeng** [comment]
@baltameme I wonder if you have the JWT token validation up `connect`. See https://github.com/raymondfeng/loopback4-example-websocket/blob/master/src/controllers/websocket.controller.ts#L18

**Comment by baltameme** [comment]
Hey @raymondfeng, yeah i've noticed that you're using websocket.controller.ts for exposing the socket connections. However, i want to enquire if its possible to following coding strategy in the following [reference](https://loopback.io/doc/en/lb4/Authentication-Tutorial.html), where i can just append the (authenticate) decorator as the following:

@ws.connect()
@authenticate('jwt')
  connect(socket: Socket) {
    console.log('Client connected: %s', this.socket.id);
    socket.join('room 1');
  }

**Comment by stale** [comment]
This issue has been marked stale because it has not seen activity within six months. If you believe this to be in error, please contact one of the code owners, listed in the `CODEOWNERS` file at the top-level of this repository. This issue will be closed within 30 days of being stale.
